node/src/types/block.rs

```diff
@@ -747,21 +759,60 @@ impl BlockHeader {
 
     /// Hash of the block header.
     pub fn hash(&self) -> BlockHash {
```

Comment:
Do we need to keep the old hash function for compatibility, or will it never be used after the upgrade?
E.g. we currently still read [the last block before the upgrade](https://github.com/CasperLabs/casper-node/blob/34c802333b88880ed7fd31f412d222dfda4519ac/node/src/effect.rs#L1568), not to take the validators directly from it, but to find out the height of the first block after the upgrade.

Reply:
That's a good question. In the CEP, I indicated that we should probably keep the old hash function, so that at least we can validate old storage. I'm not sure how critical that is, though.